pkg/cluster/config_test.go

```diff
@@ -1,244 +1,184 @@
+	})
+
+	JustAfterEach(func() {
+		_ = os.RemoveAll(configPath)
+	})
+
+	When("the CNI is kindnet", func() {
+		It("should generate the correct kind config file", func() {
+			verifyKindConfigFile(configDir, configPath, "default_cni.golden")
 		})
 	})
 
-	Context("Custom flags", func() {
-		It("Should set KubeAdminAPIVersion to kubeadm.k8s.io/v1beta1", func() {
-			usr, err := user.Current()
-			Ω(err).ShouldNot(HaveOccurred())
-
-			imageName := "kindest/node:v1.11.1"
-			cni := "weave"
-			got, err := cluster.PopulateConfig(1, imageName, cni, false, false, false, 5*time.Minute)
-			Ω(err).ShouldNot(HaveOccurred())
-			Expect(got).Should(Equal(&cluster.Config{
-				Cni:                 "weave",
-				Name:                defaults.ClusterNameBase + strconv.Itoa(1),
-				PodSubnet:           "10.4.0.0/14",
-				ServiceSubnet:       "100.1.0.0/16",
-				DNSDomain:           defaults.ClusterNameBase + strconv.Itoa(1) + ".local",
-				KubeAdminAPIVersion: "kubeadm.k8s.io/v1beta1",
-				NumWorkers:          defaults.NumWorkers,
-				KubeConfigFilePath:  filepath.Join(usr.HomeDir, ".kube", "kind-config-"+defaults.ClusterNameBase+strconv.Itoa(1)),
-				WaitForReady:        0,
-				NodeImageName:       "kindest/node:v1.11.1",
-				Retain:              false,
-				Tiller:              false,
-			}))
+	When("the CNI is custom", func() {
+		BeforeEach(func() {
+			config.Name = "custom"
+			config.Cni = "weave"
 		})
 
-		It("Should set KubeAdminAPIVersion to kubeadm.k8s.io/v1beta2", func() {
-			usr, err := user.Current()
-			Ω(err).ShouldNot(HaveOccurred())
+		It("should generate the correct kind config file", func() {
+			verifyKindConfigFile(configDir, configPath, "custom_cni.golden")
+		})
+	})
 
-			imageName := "kindest/node:v1.16.3"
-			cni := "kindnet"
-			got, err := cluster.PopulateConfig(1, imageName, cni, false, false, false, 5*time.Minute)
-			Ω(err).ShouldNot(HaveOccurred())
-			Expect(got).Should(Equal(&cluster.Config{
-				Cni:                 "kindnet",
-				Name:                defaults.ClusterNameBase + strconv.Itoa(1),
-				PodSubnet:           "10.4.0.0/14",
-				ServiceSubnet:       "100.1.0.0/16",
-				DNSDomain:           defaults.ClusterNameBase + strconv.Itoa(1) + ".local",
-				KubeAdminAPIVersion: "kubeadm.k8s.io/v1beta2",
-				NumWorkers:          defaults.NumWorkers,
-				KubeConfigFilePath:  filepath.Join(usr.HomeDir, ".kube", "kind-config-"+defaults.ClusterNameBase+strconv.Itoa(1)),
-				WaitForReady:        5 * time.Minute,
-				NodeImageName:       "kindest/node:v1.16.3",
-				Retain:              false,
-				Tiller:              false,
-			}))
+	When("the number of workers is 5", func() {
+		BeforeEach(func() {
+			config.NumWorkers = 5
+		})
+
+		It("should generate the correct kind config file", func() {
+			verifyKindConfigFile(configDir, configPath, "custom_five_workers.golden")
 		})
+	})
+}
+
+func verifyKindConfigFile(configDir string, actualConfigFilePath string, expConfigFileName string) {
```

Comment:
nit: why `expConfigFileName` and not `expectedConfigFileName`?

Reply:
A bit shorter but I'm fine either way.

Reply:
Done